src/Test/Utilities/Shared/Compilation/IRuntimeUtility.cs

```diff
@@ -0,0 +1,253 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Utilities;
+
+namespace Roslyn.Test.Utilities
+{
+    public class  RuntimeUtilityFactory
+    {
+        internal static IRuntimeUtility NewInstance(IEnumerable<ModuleData> additionalDependencies = null)
+    {
+
+#if DNX
+            return new CoreCLRRuntimeEnvironment(additionalDependencies);
```

Comment:
I thought we were going to do this through reflection rather than shared projects?